docs/dev/setup/jenkins-localvc.rst

```diff
@@ -0,0 +1,779 @@
+related steps and use the configurations provided under *Separate NGINX
+Configurations*
+
+**If you want to setup everything on your local development computer,
+ignore all NGINX related steps.** **Just make sure that you use
+unique port mappings for your Docker containers (e.g.** ``8081`` **for
+Jenkins,** ``8080`` **for Artemis)**
+
+**Prerequisites:**
+
+* `Docker <https://docs.docker.com/install>`__
+
+ Make sure that docker has enough memory (~ 6GB). To adapt it, go to ``Preferences -> Resources`` and restart Docker.
+
+.. contents:: Content of this section
+    :local:
+    :depth: 3
+
+Artemis
+^^^^^^^
+
+In order to use Artemis with Jenkins as **Continuous Integration**
+Server and LocalVC as integrated **Version Control** Server, you have to configure
+the file ``application-prod.yml`` (Production Server) or
+``application-artemis.yml`` (Local Development) accordingly. Please note
+that all values in ``<..>`` have to be configured properly. These values
+will be explained below in the corresponding sections. If you want to set up a local environment, copy the values
+below into your ``application-artemis.yml`` or ``application-local.yml`` file (the latter is recommended).
+
+.. code:: yaml
+
+   artemis:
+    course-archives-path: ./exports/courses
+    repo-clone-path: ./repos
+    repo-download-clone-path: ./repos-download
+    bcrypt-salt-rounds: 11  # The number of salt rounds for the bcrypt password hashing. Lower numbers make it faster but more unsecure and vice versa.
+                            # Please use the bcrypt benchmark tool to determine the best number of rounds for your system. https://github.com/ls1intum/bcrypt-Benchmark
+    user-management:
+        use-external: false
+        internal-admin:
+            username: artemis_admin
+            password: artemis_admin
+        accept-terms: false
+        login:
+            account-name: TUM
+    version-control:
+        url: http://172.17.0.1:8080 # `http://host.docker.internal:8080` for Windows
+        user: root
+        password: dummy # has to be set, but does not matter for LocalVC
+        build-agent-git-username: jenkins
+        build-agent-git-password: artemis_admin # choose some strong password and username (gives read access to all repositories)
+    continuous-integration:
+        user: artemis_admin
+        password: artemis_admin
+        url: http://localhost:8082
+        vcs-credentials: artemis_localvc_credentials
+        artemis-authentication-token-key: artemis_notification_plugin_token
+        artemis-authentication-token-value: artemis_admin
+        build-timeout: 30
+    git:
+        name: Artemis
+        email: artemis@xcit.tum.de
+   jenkins:
+       # only required if Artemis and Jenkins cannot communicate on their public URLs
+       # e.g., Jenkins is only available in a local container network
+       internal-urls:
+           ci-url: http://jenkins:8080
+           vcs-url: http://localhost:8080
```

Comment:
See other `vcs-url` comment

Reply:
for internal that should be artemis:8080 or something similar :)

Reply:
That's assuming Artemis runs in a container. The other uses of `172.17.0.1` imply that it would not.

Reply:
Addressed with a7828f5915b09b735b666c1dd3f584499c9acd35